golem/managers/base.py

```diff
@@ -81,6 +85,24 @@ async def run(
     async def create_batch(self) -> Batch:
         return Batch(self._activity)
 
+    @property
+    def activity(self) -> Activity:
+        return self._activity
+
+    async def get_provider_id(self):
+        if not self._provider_id:
+            proposal = await self.activity.agreement.proposal.get_data()
+            self._provider_id = proposal.issuer_id
```

Comment:
`proposal` variable name is misleading as `proposal.issuer_id` suggest that there is `issuer_id` field in the `Proposal` class. On that note, maybe those two properties should be in the `Proposal` class.

```python
    _provider_id: Optional[str] = None
    _provider_node_name: Optional[str] = None
```

Reply:
what name would you suggest for the variable? `proposal_data`?

Reply:
and as for the name, yeah, they can be there too -> my point was to have them handy on the WorkContext object so that the user of the context doesn't have to dive deep each time

Reply:
I am ok with having `async def get_provider_id(self):` helper methods but I would move the logic of retrieving and caching this data to the `Proposal`

Reply:
@lucekdudek done